substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/PreMainSupport.java

```diff
@@ -0,0 +1,296 @@
+            return false;
+        }
+
+        @Override
+        public void retransformClasses(Class<?>... classes) throws UnmodifiableClassException {
+            throw new UnsupportedOperationException("Native image doesn't support retransform class at runtime.");
+        }
+
+        @Override
+        public boolean isRedefineClassesSupported() {
+            return false;
+        }
+
+        @Override
+        public void redefineClasses(ClassDefinition... definitions) throws ClassNotFoundException, UnmodifiableClassException {
+            throw new UnsupportedOperationException("Native image doesn't support redefine class at runtime.");
+        }
+
+        @Override
+        public boolean isModifiableClass(Class<?> theClass) {
+            return false;
+        }
+
+        @Override
+        public Class<?>[] getAllLoadedClasses() {
+            ArrayList<Class<?>> userClasses = new ArrayList<>();
+            CodeInfo imageCodeInfo = CodeInfoTable.getFirstImageCodeInfo();
+            while (imageCodeInfo.isNonNull()) {
+                Class<?>[] classes = NonmovableArrays.heapCopyOfObjectArray(CodeInfoAccess.getClasses(imageCodeInfo));
+                if (classes != null) {
+                    for (Class<?> clazz : classes) {
+                        if (clazz != null) {
+                            Module module = clazz.getModule();
+                            if (module == null ||
+                                            module.getName() == null ||
+                                            !isSystemClass(module)) {
+                                userClasses.add(clazz);
+                            }
+                        }
+                    }
+                }
+                imageCodeInfo = CodeInfoAccess.getNextImageCodeInfo(imageCodeInfo);
+            }
+            userClasses.trimToSize();
+            return userClasses.toArray(new Class<?>[0]);
+        }
+
+        private static boolean isSystemClass(Module module) {
+            return systemModules.contains(module.getName());
+        }
+
+        @Override
+        public Class<?>[] getInitiatedClasses(ClassLoader loader) {
+            throw new UnsupportedOperationException("Native image has flat classloader hierarchy. Can't get classes by classloader. " +
+                            "Try to call getAllLoadedClasses() to get all loaded classes.");
+        }
+
+        @Override
+        public long getObjectSize(Object objectToSize) {
+            return -1;
+        }
+
+        @Override
+        public void appendToBootstrapClassLoaderSearch(JarFile jarfile) {
+            throw new UnsupportedOperationException("Native image doesn't support modify classloader search path at runtime.");
+        }
+
+        @Override
+        public void appendToSystemClassLoaderSearch(JarFile jarfile) {
+            throw new UnsupportedOperationException("Native image doesn't support modify classloader search path at runtime.");
```

Comment:
Similar here, as a matter of fact I would add the action sentence to all `UnsupportedOperationException`s